polyfactory/collection_extender.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import random
+from abc import ABC
+from typing import Any, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import GenericAlias  # type: ignore
```

Comment:
So this will fail in lower python versions. We have a whole around either in this repo or litestar